src/register/RegistrationFields/CountryField/CountryField.jsx

```diff
@@ -65,30 +71,27 @@ const CountryField = (props) => {
 
     const { value } = event.target;
 
-    const { countryCode, displayValue, error } = validateCountryField(
+    const { error } = validateCountryField(
       value.trim(), countryList, formatMessage(messages['empty.country.field.error']), formatMessage(messages['invalid.country.field.error']),
     );
-
-    onChangeHandler({ target: { name: 'country' } }, { countryCode, displayValue });
     handleErrorChange('country', error);
   };
 
-  const handleSelected = (value) => {
-    handleOnBlur({ target: { name: 'country', value } });
-  };
-
   const handleOnFocus = (event) => {
     handleErrorChange('country', '');
     dispatch(clearRegistrationBackendError('country'));
     onFocusHandler(event);
   };
 
   const handleOnChange = (value) => {
-    onChangeHandler({ target: { name: 'country' } }, { countryCode: '', displayValue: value });
+    onChangeHandler({ target: { name: 'country' } }, { countryCode: value.selectionId, displayValue: value.userProvidedText });
+    if (value.selectionId !== '') {
+      handleOnBlur({ target: { name: 'country', value: value.userProvidedText } });
+    }
```

Comment:
What scenario is being handled here?

Reply:
If we click the dropdown icon and select the item from list that mean we are not focusing the input box we just use the dropdown icon and select the element

Reply:
isn't the `selectionId` provided on item click? i wanted to understand this check expecially.